runner/sidecar/sidecar.go

```diff
@@ -216,73 +201,12 @@ func Exec(ctx context.Context) error {
 		return err
 	}
 
-	go wait.JitterUntil(func() { defer runtimeutil.HandleCrash(); patchStepStatus() }, updateInterval, 1.2, true, ctx.Done())
-
 	ready = true
 	logger.Info("ready")
 	<-ctx.Done()
 	return nil
 }
 
-func patchStepStatusHook(context.Context) error {
-	patchStepStatus()
-	return nil
-}
-
-func patchStepStatus() {
-	ctx := context.Background()
-	patchMu.Lock()
-	defer patchMu.Unlock()
-	if ready { // don't run these if we are not ready, pollutes the logs with errors
-		for _, f := range prePatchHooks {
-			n := sharedutil.GetFuncName(f)
-			logger.Info("executing pre-patch hook", "func", n)
-			if err := f(ctx); err != nil {
-				logger.Error(err, "failed to execute hook", "func", n)
-			}
-		}
-	}
-	mu.RLock()
-	notEqual, patch := sharedutil.NotEqual(dfv1.Step{Status: lastStep.Status}, dfv1.Step{Status: step.Status})
-	mu.RUnlock()
-	if notEqual {
-		logger.Info("patching step status", "patch", patch)
-		if un, err := dynamicInterface.
-			Resource(dfv1.StepGroupVersionResource).
-			Namespace(namespace).
-			Patch(
```

Comment:
you can also remove service account `pipeline``s permission to patch step/status too